front/pages/w/[wId]/builder/assistants/create.tsx

```diff
@@ -98,6 +98,29 @@ export default function CreateAssistant({
     });
   }, [assistantTemplates]);
 
+  const openTemplateModal = useCallback(
+    async (templateId: string) => {
```

Comment:
Not a big fan of having to keep state and router aligned, why not fully relying on the router?

Reply:
Because it was not working, you can see on prod that if you open multiple templates modals then you cannot close the page modal. I think this is cleaner as is, when we open the modal we set a state and shallow push the param in the url, and when we close we clean the state and the query param.

Reply:
Seen IRL.